{{cookiecutter.project_name}}/contracts/NFT.vy

```diff
@@ -301,6 +317,49 @@ def royaltyInfo(_tokenId: uint256, _salePrice: uint256) -> (address, uint256):
     return self.owner, royalty
 {%- endif %}
 
+{%- if cookiecutter.force_royalties == 'y' %}
+# Helper functions in case market place does not support royalties to execute by this contract with _deductRoyalties()
+@internal
+@view
+def _royaltyInfo(_tokenId: uint256, _salePrice: uint256) -> (address, uint256):
+    """
+    /// @notice Called with the sale price to determine how much royalty
+    //          is owed and to whom. Important; Not all marketplaces respect this, e.g. OpenSea
+    /// @param _tokenId - the NFT asset queried for royalty information
+    /// @param _salePrice - the sale price of the NFT asset specified by _tokenId
+    /// @return receiver - address of who should be sent the royalty payment
+    /// @return owner address and royaltyAmount - the royalty payment amount for _salePrice
+    """
+
+    royalty: uint256 = convert(convert(_salePrice, decimal) * ROYALTY_TO_APPLY_TO_PRICE, uint256) # Percentage that accepts decimals
+    return self.owner, royalty
+
+@internal
+def royaltyChecker(tokenId: uint256):
```

Comment:
```suggestion
def _enforceRoyalties(caller: address, tokenId: uint256):
```

Reply:
@fubuloubu why include the `caller` if you don't use it on the function?

Reply:
In a future scenario you might want to use it to detect which marketplace is triggering the call to update the minimum royalty amount

For now you can probably skip the check if `msg.sender` is not a contract